src/coreclr/src/jit/gentree.cpp

```diff
@@ -4431,6 +4302,121 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
             }
             break;
 
+#ifdef FEATURE_SIMD
+        case GT_SIMD:
+            if (tree->AsSIMD()->IsBinary())
+            {
+                GenTree* op1 = tree->AsSIMD()->GetOp(0);
+                GenTree* op2 = tree->AsSIMD()->GetOp(1);
+
+                if (tree->IsReverseOp())
+                {
+                    std::swap(op1, op2);
+                }
+
+                level = gtSetEvalOrder(op1);
+                lvl2  = gtSetEvalOrder(op2);
+
+                if ((fgOrder == FGOrderTree) && (level < lvl2) && gtCanSwapOrder(op1, op2))
+                {
+                    tree->gtFlags ^= GTF_REVERSE_OPS;
+                    std::swap(level, lvl2);
+                }
+
+                if (level == 0)
+                {
+                    level = lvl2;
+                }
+                else if (level == lvl2)
+                {
+                    level += 1;
+                }
+
+                costEx = op1->GetCostEx() + op2->GetCostEx() + 1;
+                costSz = op1->GetCostSz() + op2->GetCostSz() + 1;
+            }
+            else
+            {
+                level  = 0;
+                costEx = 1;
+                costSz = 1;
+
+                for (GenTreeSIMD::Use& use : tree->AsSIMD()->Uses())
+                {
+                    level = max(level, gtSetEvalOrder(use.GetNode()));
```

Comment:
Need to double check this, it's not equivalent to the old `gtSetListOrder` code. On the other hand it's not clear if what `gtSetListOrder` did makes sense for `SIMDIntrinsicInitN`. Also, when looking more closely at the way `SIMDIntrinsicInitN` and `gtSetEvalOrder` work it seems that things are quite messy:
* `SIMDIntrinsicInitN` has horrible register requirements - up to 5 registers - because it needs to first evaluate all operands and then it stitches everything together. This approach will never fly if we try to make intrinsics for `Create` methods like `Vector128<byte> Create(byte e0, byte e1, byte e2, byte e3, byte e4, byte e5, byte e6, byte e7, byte e8, byte e9, byte e10, byte e11, byte e12, byte e13, byte e14, byte e15)`...
* `gtSetEvalOrder` runs before other optimizations so it has no idea that some trees may turn into constants or variables thus significantly reducing register requirements. This includes `SIMDIntrinsicInitN` becoming a pseudo-constant in lowering...
* We can't control the evaluation order for nodes with more than 2 operands unless we add more information to `GenTreeSIMD`. And we're more or less out of space for any new information (e.g. an array of integers that would indicate the execution order of eveyr operand).

Reply:
I tweaked "level" and costs a bit but "level" is still different from what `gtSetListOrder` does. But what `gtSetListOrder` does is dubious anyway:
https://github.com/dotnet/runtime/blob/c1a51e2c538a4c520458190d9c4a1645fc375525/src/coreclr/src/jit/gentree.cpp#L2522-L2533 Sethi-Ullman number is basically `(l1 == l2) ? (l1 + 1) : max(l1, l2)` and there's no trace of `max` anywhere in `gtSetListOrder`. I suspect `max` was supposed to be result of setting `GTF_REVERSE_OPS` and swapping levels:
https://github.com/dotnet/runtime/blob/c1a51e2c538a4c520458190d9c4a1645fc375525/src/coreclr/src/jit/gentree.cpp#L2501-L2508 Except that `GTF_REVERSE_OPS` is never set on list nodes.
Whatever. Doesn't really matter and `gtSetEvalOrder` is a mess anyway.